api/v1alpha2/types.go

```diff
@@ -147,9 +150,9 @@ var (
 
 // SecurityGroup defines an Azure security group.
 type SecurityGroup struct {
-	ID           string       `json:"id"`
+	ID           string       `json:"id,omitempty"`
 	Name         string       `json:"name"`
```

Comment:
omitempty here too?

Reply:
my thinking was: if security group is specified, then it should at least have a name" but I guess it's also valid to provide an empty security group. wdyt?

Reply:
We can always use a defaulting webhook to generate the name if none is provided.